pkg/odo/cli/service/create.go

```diff
@@ -237,76 +274,105 @@ func (o *ServiceCreateOptions) Validate() (err error) {
 			if err != nil {
 				return err
 			}
-			// var jsonCR map[string]interface{}
-			err = yaml.Unmarshal(fileContents, &o.CustomResourceDefinition)
+
+			err = yaml.Unmarshal(fileContents, &d.OriginalCRD)
 			if err != nil {
 				return err
 			}
 
 			// Check if the operator and the CR exist on cluster
-			o.CustomResource = o.CustomResourceDefinition["kind"].(string)
-			csvs, err := o.KClient.GetClusterServiceVersionList()
+			var csv olm.ClusterServiceVersion
+			o.CustomResource, csv, err = svc.GetCSV(o.KClient, d.OriginalCRD)
 			if err != nil {
 				return err
 			}
 
-			csv, err := doesCRExist(o.CustomResource, csvs)
+			// all is well, let's populate the fields required for creating operator backed service
+			o.group, o.version, o.resource, err = svc.GetGVRFromOperator(csv, o.CustomResource)
 			if err != nil {
-				return fmt.Errorf("Could not find specified service/custom resource: %s\nPlease check the \"kind\" field in the yaml (it's case-sensitive)", o.CustomResource)
+				return err
 			}
 
-			// all is well, let's populate the fields required for creating operator backed service
-			o.group, o.version = groupVersionALMExample(o.CustomResourceDefinition)
-			o.resource = resourceFromCSV(csv, o.CustomResource)
-			o.ServiceName, err = serviceNameFromCRD(o.CustomResourceDefinition, o.ServiceName)
-			return err
-		}
-		if o.CustomResource != "" {
-			// make sure that CSV of the specified ServiceType exists
-			csv, err := o.KClient.GetClusterServiceVersion(o.ServiceType)
+			err = d.validateMetadataInCRD()
 			if err != nil {
-				// error only occurs when OperatorHub is not installed.
-				// k8s does't have it installed by default but OCP does
 				return err
 			}
 
-			var almExamples []map[string]interface{}
-			val, ok := csv.Annotations["alm-examples"]
-			if ok {
-				err = json.Unmarshal([]byte(val), &almExamples)
-				if err != nil {
-					return errors.Wrap(err, "unable to unmarshal alm-examples")
+			if o.ServiceName != "" && !o.DryRun {
+				// First check if service with provided name already exists
+				svcFullName := strings.Join([]string{o.CustomResource, o.ServiceName}, "/")
+				exists, err := svc.OperatorSvcExists(o.KClient, svcFullName)
+				if kerrors.IsNotFound(err) {
```

Comment:
I think this should be the opposite. If the error is a `not found` error, we should continue with exists as false.

Reply:
I checked again and the opposite condition doesn't work. We can return just false and no error from OperatorSvcExists if the resource is not found. I think that is sufficient to let the caller know about the existence. In that case we won't need the kerrors package too.